src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/payment-history/PaymentHistory.jsx

```diff
@@ -0,0 +1,229 @@
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Box,
+  Button,
+  InputLabel,
+  MenuItem,
+  Select,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { useState, useCallback } from 'react';
+
+import creditCard from '../../../../../../../../assets/icons/credit-card.png';
+import commonStyles from '../style.module.css';
+import styles from './PaymentHistory.module.css';
+import tableStyles from '../../../table/PayableTable.module.css';
+import { generateDate } from '../../../../../../../../utils';
+import { NoData, LoadingSpinner, TooltipText } from '../../../../../../../../components';
+import { TableBodyWrapper, TableCellWrapper } from '../../../table/PayableTable';
+
+const inputTypes = {
+  text: 'text',
+  price: 'price',
+  date: 'date',
+  select: 'select',
+  file: 'file',
+};
+
+const columns = [
+  { id: 'invoiceId', label: 'Invoice ID' },
+  { id: 'paidAmount', label: 'Paid Amount' },
+  { id: 'paymentDate', label: 'Payment Date' },
+  { id: 'paymentType', label: 'Payment Type' },
+  { id: 'attachment', label: 'Attachment' },
+];
+
+const EditableTableCell = ({ rowValue, type, inputLabel }) => {
+  const [input, setInput] = useState(rowValue);
+
+  return (
+    <TableCellWrapper>
+      {(() => {
+        switch (type) {
```

Comment:
maybe we just need to write a mapper, like: 
```
const mapToTypes = {
  [inputTypes.text]: <TextField
                label={inputLabel}
                size="small"
                variant="outlined"
                value={input}
                onChange={(e) => setInput(e.target.value)}
              />
               ....
}
```

and avoid switch case?

Reply:
This code part will be changed. This was simple and fastest way to create some ui view